.github/workflows/check.yaml

```diff
@@ -144,24 +144,28 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        runs-on: [[ubuntu-24.04], [Ubuntu_ARM64_4C_16G_01]]
+        runs-on: [[ubuntu-24.04], [arm64, noble, self-hosted], [s390x, noble, self-hosted], [ppc64el, noble, self-hosted]]
         test-command: ['tox -e func']
-        juju-channel: ["3.4/stable"]
+        juju-channel: ["3.6/stable"]
     steps:
       - uses: actions/checkout@v4
         with:
           submodules: true
 
-        # arm64 runners don't have gcc installed by default
       - name: Install dependencies
         run: |
           sudo apt update
-          sudo apt install -y gcc
 
-      - name: Setup Python
-        uses: actions/setup-python@v5
-        with:
-          python-version: "3.12"
+          # s390x runners don't have a rust toolchain available
+          # NOTE: only required for building from source bcrypt>=4.0.0 or cryptography>=3.4
+          sudo apt install -y cargo
+          # also for building cryptography>=3.4, on s390x runners
+          sudo apt install -y libssl-dev
+          sudo apt install -y pkg-config
+
+      # - name: Update system PATH for running pipx software
+      #   run: |
+      #     echo "$HOME/.local/bin" >> "$GITHUB_PATH"
```

Comment:
This can be deleted - this step was only required for the enterprise arm64 runner (now the setup-python step is gone). But I'm sure we'll be wanting to move away from that runner for all our projects (to use either our self hosted runners or the new public free github-hosted arm64 runners).